python/cudf/cudf/core/reshape.py

```diff
@@ -13,7 +13,34 @@
 _axis_map = {0: 0, 1: 1, "index": 0, "columns": 1}
 
 
-def concat(objs, axis=0, ignore_index=False, sort=None):
+def align_objs(objs):
+    """Align a set of Series or Dataframe objects.
+
+    Parameters
+    ----------
+    objs : list of DataFrame, Series, or Index
+
+    Returns
+    -------
+    A set of reindexed and aligned objects ready for concatenation
+    """
+    index = objs[0].index
+    for obj in objs[1:]:
+        name = index.name
+        index = (
+            cudf.DataFrame(index=obj.index)
+            .join(cudf.DataFrame(index=index), how="outer")
+            .index
+        )
+        index.name = name
+
+    for i, obj in enumerate(objs):
+        objs[i] = obj.reindex(index)
+
+    return objs
+
+
+def concat(objs, axis=0, ignore_index=False, sort=None, align_index=False):
```

Comment:
Instead of `align_index` follow the Pandas API: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html

Should always align unless `ignore_index=True`.